src/mirror.c

```diff
@@ -221,10 +244,14 @@ void handle_mirror_if_stale(void)
 	 * will be used due to the diff being negative. */
 	int diff = central_version - mirror_version;
```

Comment:
counting in this diff to decide if we get in or not is very fragile.
If we have errors on central_version and mirror_versions we're just getting into the if if mirror_version error is -1. But in this case I don't think it makes sense to unset the mirror, because we can have a connection problems.

So if the central_version is negative we should just ignore.
If the mirror_version has error or it's outdated and central_version works, we should unset the mirror.

Reply:
ok I'll refactor that

Reply:
done